### PR TITLE
kie-issues#1213: configure for dockerhub registry

### PR DESCRIPTION
**Pull request by jstastny-cz**
Fixes:
* apache/incubator-kie-issues#1213

Depends on `docker.io/apache/incubator-kie-kogito-ci-build:main-latest` being released first (after #1197 is merged and pipeline execution succeeds).

Part of PR group:
* apache/incubator-kie-kogito-pipelines#1198
* apache/incubator-kie-drools#5952
* apache/incubator-kie-optaplanner#3087

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

File: .ci/jenkins/config/branch.yaml [hunk 1]
```diff
@@ -79,16 +79,16 @@ maven:
     creds_id: apache-nexus-kie-deploy-credentials
 cloud:
   image:
-    registry_credentials: tradisso_registry_token # TODO set to `kogito-quay-token`
-    registry: quay.io
-    namespace: tradisso
+    registry_credentials: dockerhub_apache_kie_registry_token
```
Comment (rodrigonull) [inline .ci/jenkins/config/branch.yaml:82]:
```suggestion
    registry_user_credentials_id: DOCKERHUB_USER
    registry_token_credentials_id: DOCKERHUB_TOKEN
```

**Review by rodrigonull — CHANGES_REQUESTED** [review]

**Review by rodrigonull — APPROVED** [review]
LGTM . Thanks!

**Comment by rodrigonull** [comment]
@jstastny-cz I'm not sure why the dsl test is failing. Can you verify?

**Review by ricardozanini — APPROVED** [review]

File: .ci/jenkins/Jenkinsfile.promote [hunk 2]
```diff
@@ -316,10 +316,8 @@ void pushOperatorFinalImage() {
...
-    if (!imageUtils.removeQuayTag(getOperatorImageNamespace(), getOperatorImageName(), getOperatorImageTemporaryTag())) {
-        String temporaryImageName = imageUtils.getImageFullName(getOperatorImageNamespace(), getOperatorImageName(),
-                getOperatorImageTemporaryTag())
-        error "Cannot remove the OptaPlanner Operator temporary image tag (${temporaryImageName}) from quay.io. "
-                + "The tag should be removed manually."
-    }
+    String temporaryImageName = imageUtils.getImageFullName(getOperatorImageNamespace(), getOperatorImageName(),
+            getOperatorImageTemporaryTag())
+    // https://docs.docker.com/docker-hub/api/latest/#tag/repositories
+    error "Dockerhub API does not allow deleting tags via API. The OptaPlanner Operator temporary image tag (${temporaryImageName}) should be removed manually."
```
Comment (ricardozanini) [inline .ci/jenkins/Jenkinsfile.promote:322]: @rodrigonull maybe it's this line the CI is complaining about?

>     org.spockframework.runtime.UnallowedExceptionThrownError at JobScriptsSpec.groovy:97

Comment (ricardozanini) [reply .ci/jenkins/Jenkinsfile.promote:322]: I guess we can use `throw`? https://www.jenkins.io/doc/book/pipeline/syntax/#flow-control

I'm not sure, I'm not a Jenkins SME just brainstorming. :)

**Comment by jstastny-cz** [comment]
Those tests don't work well when my fork name is different than the actual repo :/

**Comment by jstastny-cz** [comment]
Tests passed when running locally with correct overrides. The test scripts would need much time investment, I might rather rename my forks :-D

**Pull request merged**